src/sequence_batch_scheduler/sequence_batch_scheduler.cc

```diff
@@ -884,6 +884,24 @@ SequenceBatchScheduler::ReleaseSequenceSlot(
 {
   std::unique_lock<std::mutex> lock(mu_);
 
+  // If we are releasing the slot for a cancelled sequence,
+  // we have to clean up the sequence
+  // otherwise the reaper will try to clean it up again.
+  if (!requests->empty() && requests->front()) {
+    const InferenceRequest::SequenceId& corr_id =
+        requests->front()->CorrelationId();
```

Comment:
@kthui is there a possiblity for requests other than the front request to be cancelled?

Reply:
I believe cancelling any request in the sequence should cancel the entire sequence. The `requests->front()` check is just to make sure that the request object is not null (a null request is inserted by reaper thread to signal a timed out sequence). Otherwise I'll let @kthui comment if you're asking something else.